grouper/ctl/user.py

```diff
@@ -127,41 +123,100 @@ def user_command(args):
         )
 
 
-def add_parser(subparsers):
-    user_parser = subparsers.add_parser("user", help="Edit user")
-    user_parser.set_defaults(func=handle_command)
-    user_subparser = user_parser.add_subparsers(dest="subcommand")
-
-    user_subparser.add_parser("list", help="List all users and their account statuses")
-
-    user_create_parser = user_subparser.add_parser("create", help="Create a new user account")
-    user_create_parser.add_argument("username", nargs="+")
-    user_create_parser.add_argument(
-        "--role-user",
-        default=False,
-        action="store_true",
-        help="If given, identifies user as a role user.",
-    )
-
-    user_disable_parser = user_subparser.add_parser("disable", help="Disable a user account")
-    user_disable_parser.add_argument("username", nargs="+")
-
-    user_enable_parser = user_subparser.add_parser("enable", help="(Re-)enable a user account")
-    user_enable_parser.add_argument("username", nargs="+")
-    user_enable_parser.add_argument(
-        "--preserve-membership",
-        default=False,
-        action="store_true",
-        help="Unless provided, scrub all group memberships when re-enabling user.",
-    )
-
-    user_key_parser = user_subparser.add_parser("add_public_key", help="Add public key to user")
-    user_key_parser.add_argument("username")
-    user_key_parser.add_argument("public_key")
-
-    user_set_metadata_parser = user_subparser.add_parser(
-        "set_metadata", help="Set metadata on user"
-    )
-    user_set_metadata_parser.add_argument("username")
-    user_set_metadata_parser.add_argument("metadata_key")
-    user_set_metadata_parser.add_argument("metadata_value")
+class UserCommand(CtlCommand, ConvertUserToServiceAccountUI):
+    """Commands to modify users."""
+
+    @staticmethod
+    def add_arguments(parser):
+        # type: (ArgumentParser) -> None
+        parser.add_argument(
+            "-a",
+            "--actor",
+            required=False,
+            dest="actor_name",
+            help=(
+                "Name of the entity performing this action."
+                " Must be a valid Grouper human or service account."
+            ),
+        )
+
+        subparser = parser.add_subparsers(dest="subcommand")
+
+        user_key_parser = subparser.add_parser("add_public_key", help="Add public key to user")
+        user_key_parser.add_argument("username")
+        user_key_parser.add_argument("public_key")
```

Comment:
Do we ideas for how to keep/make this configuration readable in a world where there are a ton more plugins? At the least we have backlog tasks to add a few more, and this is already a bit painful to visually scan

Reply:
Yeah, I think we should add another layer that looks like the subcommand layer and configures and runs each sub-subcommand. I can put up another diff to look at that implements that.